src/libraries/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -55,6 +56,7 @@ internal ZipArchiveEntry(ZipArchive archive, ZipCentralDirectoryFileHeader cd)
             _versionMadeBySpecification = (ZipVersionNeededValues)cd.VersionMadeBySpecification;
             _versionToExtract = (ZipVersionNeededValues)cd.VersionNeededToExtract;
             _generalPurposeBitFlag = (BitFlagValues)cd.GeneralPurposeBitFlag;
+            _isEncrypted = _generalPurposeBitFlag.HasFlag(BitFlagValues.IsEncrypted);
```

Comment:
This is faster than using `Enum.HasFlag`. See [assembly size in sharplab.io](https://sharplab.io/#v2:C4LghgzgtgNAJiA1AHwAICYAMBYAUBgRjzwDcwAnAAjIBtKBeSgIQEtgAxGsAcwDUwaAVwCmEAHQBJCAFEAdgGNyATwAOwYXErJmbTj35DRYgCJhgYY6MUs1Ae3IBuPKgIBOABS0AlE9zF8bu4ARACyBCCUQZSIlGGeAl4+zoGh6BFRMSHo8TSJvngARra2dHGsHFx8AiIQlNzCssLkAgAKguQqthDC5XrcXgwAfHUNTa3tnd29lWIAEpB97tP61UZScoqq6nBJfrhFJbHZy1WGtfWNzTRtHV09upUD9MPuF2PXE3cnlABkOhUrM6SGQKZRqDQDACEjEw+VwAG0+hAALp4BqCKD/PoGGqUCKCCAAC3swEoAG9KOtQVsNAxKJgAB4EGCUUzmSwQax2KgwhkADhZAFVZCx5LY4MJ2CwaMIAHJgKDCACCsjgAGFbFBFbJSby+ZhMJQAL54IA===).
```suggestion
            _isEncrypted = (_generalPurposeBitFlag & BitFlagValues.IsEncrypted) != 0;
```